app/dashboard/utils.py

```diff
@@ -1087,3 +1087,18 @@ def get_token_recipient_senders(network, recipient_address, token_address):
     )
 
     return [process_log(log) for log in logs]
+
+
+def get_hackathon_event(title, event, network):
```

Comment:
@owocki  It's this which is eats up the loadtime from the looks of it ! 
Cause we invoke this tice

Reply:
makes sense. i think persisting this info in the model will solve muhc of the issues

Reply:
and i defer to the engineers but i think a limit on the # of db calls per pageload is a good idea, as a code standsrad

Reply:
> makes sense. i think persisting this info in the model will solve muhc of the issues

I would like to help resolve this @thelostone-mc @owocki. However, I have yet to work with the presistance methods in Gitcoin project. With that said, are you basically looking for the `event_bounties` and `registrants` in the `get_hackathon_event(title, event, network)` function to be cached using the `.cache()` method from the installed django-cacheops?

Reply:
> makes sense. i think persisting this info in the model will solve muhc of the issues

I would like to help resolve this @thelostone-mc @owocki. However, I have yet to work with the presistance methods in Gitcoin project. With that said, are you basically looking for the `event_bounties` and `registrants` querysets in the `get_hackathon_event(title, event, network)` function to be cached using the `.cache()` method from the installed django-cacheops?

Reply:
> makes sense. i think persisting this info in the model will solve muhc of the issues

I would like to help resolve this @thelostone-mc @owocki. However, I have yet to work with the presistance methods in Gitcoin project. With that said, are you basically looking for the `event_bounties` and `registrants` querysets in the `get_hackathon_event(title, event, network)` function to be cached using the `.cache()` method from the installed django-cacheops? Or perhaps caching the `get_hackathon_event` function?

Reply:
i'd just throw them in a new JSONField on the object, and then make sure the `create_page_cache` mgmt command refreshes them whenever its fun

the `.cache()` thing only works if you have something that periodically warms the cache, so its not ideal

Reply:
@walidmujahid any update? the page is still loading in over 2 seconds and will get worse with every hack we add

Reply:
@thelostone-mc @owocki I had missed the fact that the `event_bounties` and `registrants` have now become obsolete and are currently unused. I have removed them -draft PR https://github.com/gitcoinco/web/pull/7254 -, however, I am not sure if this helps resolve the main loadtime issue seen as I have yet to locally test this. 

On the matter of persisting the conetxt or certain information for the context, I believed I had the solution, however, I am a bit uncertain about it. The information in the `get_hackathon_event` util function includes a badly named  key called `title` -which helps categorise the hackathon event based on whether it is current, upcoming, or finished- and `hackathon` which is just a querystring. 

What I am hesitant about is putting `title` in a JSONField in the object, and then refreshing whether it is `current`, or `finished`, or otherwise because, if I do, would the cron delay for running `create_page_cache` cause some confusion? Is the cron delay something to be concerned about in the first place? If yes then would putting all finished hackathons in an `JSONStore` in `create_page_cache` help solve the laodtime issue? And if the cron delay is not much of an issue, the would it make sense to just go ahead and have `create_page_cache` store the entire context in a `JSONStore` and as such newly added hackathons or updates will appear whenever the cron runs?

If I am overcomplicating things, forgive me and let me know.

Reply:
i am willing to settle for a slight delay in updating the hack info if it means the page loads in 0.5s instead of 2.5s.  i say proceed with the caching approach!

pls invoice me gitcoin.co/requestmoney when u have a PR out

Reply:
Since PR https://github.com/gitcoinco/web/pull/7254 has been merge I am going to set this conversation as resolved.